### DISCUSSION TITLE
Logging object as json

### DISCUSSION THREAD

**Discussion by mochocki** [klogging/klogging#297] [Q&A] [answered]
I would like to have object passed to logger rendered as json - is that possible?
Currently I'm doing something like this:
```
 var obj2log = Photo(cameraMake = "cameramake", exposureTime = "exposuretime", focalLength = 1.0)
logger.info { "$objectToLog" }
```
It is being logged as
`... "objectToLog":"Photo(cameraMake=cameramake, cameraModel=null, focalLength=1.0, apertureFNumber=null, isoEquivalent=null, exposureTime=exposuretime)" ...
`
instead of just logging it as json.

**Comment by mjstrasser** [comment, marked as answer]
Hi @mochocki, thanks for your question.

Unfortunately there is no simple way in Kotlin to simply render a data class as JSON, unlike JavaScript. 

I could not find a way to get Klogging to recognise at runtime a class annotated with `@Serializable` and render it as JSON.
